src/memory_targets/memory_target_numa.c

```diff
@@ -25,12 +26,18 @@ static umf_result_t numa_initialize(void *params, void **memTarget) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
+    int rv = umfMTLinearBaseAllocatorCreate();
+    if (rv) {
+        return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+    }
+
     struct umf_numa_memory_target_config_t *config =
         (struct umf_numa_memory_target_config_t *)params;
 
     struct numa_memory_target_t *numaTarget =
-        malloc(sizeof(struct numa_memory_target_t));
+        umf_ba_linear_alloc(MT_LBA_pool, sizeof(struct numa_memory_target_t));
```

Comment:
Hm, actually for memspace and memory targets we should also not use linear_alloc I think. Memspaces and memory targets can be created by users at any time (and destroyed). Can we use the same BA_pool as for pools and providers?

Reply:
I think there could be an issue with allocating array of memory_targets in a memspace (if there is more than `SIZE_BA_POOL_CHUNK / sizeof (struct memory_target_t)` targets) - I think we should replace the array with some list or segment array implementation where each segment is SIZE_BA_POOL_CHUNK in size. But perhaps we can do that in a separate PR?